src/devices/Pcd8544/Pcd8544.cs

```diff
@@ -0,0 +1,487 @@
+        /// Draw a point
+        /// </summary>
+        /// <param name="x">The X coordinate.</param>
+        /// <param name="y">The Y coordinate.</param>
+        /// <param name="isOn">True if the point has pixels on, false for off.</param>
+        /// <returns>True if success</returns>
+        public bool DrawPoint(int x, int y, bool isOn)
+        {
+            if (x < 0 || x >= 84 || y < 0 || y >= 48)
+            {
+                return false;
+            }
+
+            int index = ((x % 84) + (int)(y * 0.125) * 84);
+
+            byte bitMask = (byte)(1 << (y % 8));
+
+            if (isOn)
+            {
+                _byteMap[index] |= bitMask;
+            }
+            else
+            {
+                _byteMap[index] &= (byte)~bitMask;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Draw a point
+        /// </summary>
+        /// <param name="point">The point to draw.</param>
+        /// <param name="isOn">True if the point has pixels on, false for off.</param>
+        /// <returns></returns>
+        public bool DrawPoint(Point point, bool isOn) => DrawPoint(point.X, point.Y, isOn);
+
+        /// <summary>
+        /// Draw a line
+        /// </summary>
+        /// <param name="x1">The first point X coordinate.</param>
+        /// <param name="y1">The first point Y coordinate.</param>
+        /// <param name="x2">The second point X coordinate.</param>
+        /// <param name="y2">The second point Y coordinate.</param>
+        /// <param name="isOn">True if the line has pixels on, false for off.</param>
+        public void DrawLine(int x1, int y1, int x2, int y2, bool isOn)
+        {
+            // This is a common line drawing algorithm. Read about it here:
+            // http://en.wikipedia.org/wiki/Bresenham's_line_algorithm
+            int sx = (x1 < x2) ? 1 : -1;
+            int sy = (y1 < y2) ? 1 : -1;
+
+            int dx = x2 > x1 ? x2 - x1 : x1 - x2;
+            int dy = y2 > x1 ? y2 - y1 : y1 - y2;
+
+            float err = dx - dy, e2;
+
+            // if there is an error with drawing a point or the line is finished get out of the loop!
+            while (!((x1 == x2 && y1 == y2) || !DrawPoint(x1, y1, isOn)))
+            {
+                e2 = 2 * err;
+
+                if (e2 > -dy)
+                {
+                    err -= dy;
+                    x1 += sx;
+                }
+
+                if (e2 < dx)
+                {
+                    err += dx;
+                    y1 += sy;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Draw a line
+        /// </summary>
+        /// <param name="p1">First point coordinate.</param>
+        /// <param name="p2">Second point coordinate.</param>
+        /// <param name="isOn">True if the line has pixels on, false for off.</param>
+        public void DrawLine(Point p1, Point p2, bool isOn) => DrawLine(p1.X, p1.Y, p2.X, p2.Y, isOn);
```

Comment:
I'd suggest only this API to be public.

Reply:
It doesn't really cost anything to have few more options.

Reply:
fair enough, we can clean it up once we get the actual abstraction